### PR TITLE
Linting with clang-tidy

### LINKED ISSUE #978: [CI] Integrate lint
clang-tidy or cppcheck

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Add linting infrastructure with clang-tidy.  Include some lint fixes.  Fixes #978 .

## Testing instructions

Standard unit tests.  New linux-lint CI build.

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by catenocrypt** [comment]
I consider this done.
I plan to keep it as a CI build executed on PRs, but make it not Required.

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,6 +57,17 @@ if(CODE_COVERAGE AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
     set(CMAKE_EXE_LINKER_FLAGS "-fprofile-arcs -ftest-coverage")
 endif()
 
+option(CLANG_TIDY "Enable cppcheck static code analysis" OFF)
```
Comment (hewigovens) [inline CMakeLists.txt:60]:
```suggestion
option(CLANG_TIDY "Enable clang-tidy static code analysis" OFF)
```

File: .github/workflows/linux-lint.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Linux Lint
```
Comment (hewigovens) [inline .github/workflows/linux-lint.yaml:1]: Let's merge it with `linux-ci.yml` and only run on master (it takes more than a hour)

Comment (catenocrypt) [reply .github/workflows/linux-lint.yaml:1]: Linux-ci is sill needed on PRs for feedback -- checking build and coverage.  Linting can be separate, but run only on master.

**Comment by catenocrypt** [comment]
Renamed to Linux-all, run on master, contains build, lint, test, coverage.
Linux-CI is run on PRs only (not on master), contains build, test, coverage, but no lint

File: .github/workflows/linux-lint.yaml (hunk 2 above)
Comment (catenocrypt) [reply .github/workflows/linux-lint.yaml]: Combined into one workflow file

**Review by hewigovens — APPROVED** [review]

**Review by vikmeup — APPROVED** [review]

**Pull request merged**